misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JDIEventProcessor.java

```diff
@@ -103,11 +104,31 @@ private boolean processEvent(EventSet eventSet, Event event) {
             configureUserBreakPoints(evt.referenceType());
             eventSet.resume();
         } else if (event instanceof BreakpointEvent) {
-            StoppedEventArguments stoppedEventArguments = new StoppedEventArguments();
-            stoppedEventArguments.setReason(StoppedEventArgumentsReason.BREAKPOINT);
-            stoppedEventArguments.setThreadId(((BreakpointEvent) event).thread().uniqueID());
-            stoppedEventArguments.setAllThreadsStopped(true);
-            context.getClient().stopped(stoppedEventArguments);
+            Value value = null;
+            String qualifiedClassName =
+                getQualifiedClassName(context, ((BreakpointEvent) event).location().declaringType());
+            Map<Integer, BalBreakpoint> balBreakpoints = this.balBreakpoints.get(qualifiedClassName);
+            int lineNumber = ((BreakpointEvent) event).location().lineNumber();
+
+            if (balBreakpoints != null && balBreakpoints.containsKey(lineNumber)) {
+                value = context.getAdapter().evaluateExpression(balBreakpoints.get(lineNumber).getCondition(),
+                    ((BreakpointEvent) event).thread());
+            }
+            if (balBreakpoints != null && balBreakpoints.containsKey(lineNumber)
```

Comment:
Is this null check for `balBreakpoints` necessary in here?

Reply:
Refer to [this](https://github.com/ballerina-platform/ballerina-lang/pull/28968#discussion_r586292855) comment.